parser/files/reading.go

```diff
@@ -0,0 +1,463 @@
+			line := strings.Fields(fileScanner.Text())
+			switch line[0][1:] {
+			case "separator":
+				var err error
+				toReturn.Separator, err = strconv.Unquote("\"" + line[1] + "\"")
+				if err != nil {
+					return toReturn, err
+				}
+			case "set_separator":
+				toReturn.SetSep = line[1]
+			case "empty_field":
+				toReturn.Empty = line[1]
+			case "unset_field":
+				toReturn.Unset = line[1]
+			case "fields":
+				toReturn.Names = line[1:]
+			case "types":
+				toReturn.Types = line[1:]
+			case "path":
+				toReturn.ObjType = line[1]
+			}
+		} else {
+			//We are done parsing the comments
+			break
+		}
+	}
+
+	if len(toReturn.Names) != len(toReturn.Types) {
+		return toReturn, errors.New("name / type mismatch")
+	}
+	return toReturn, nil
+}
+
+func mapZeekHeaderToParseType(header *BroHeader, broDataFactory func() pt.BroData, logger *log.Logger) (ZeekHeaderIndexMap, error) {
+	broData := broDataFactory()
+	structType := reflect.TypeOf(broData).Elem()
+
+	indexMap := ZeekHeaderIndexMap{
+		NthLogFieldExistsInParseType: make([]bool, len(header.Names)),
+		NthLogFieldParseTypeOffset:   make([]int, len(header.Names)),
+	}
+
+	// parseTypeFieldInfo and the parseTypeFields map record the names, types, and offsets of the
+	// Zeek fields we want to populate the broData with. Recording this info in a map allows
+	// us to match the Zeek header to the parse type fields without nested loops.
+	type parseTypeFieldInfo struct {
+		zeekType             string
+		parseTypeFieldOffset int
+	}
+	// parseTypeFields maps from Zeek field names to the associated info as defined by the
+	// broData struct tags
+	parseTypeFields := make(map[string]parseTypeFieldInfo)
+
+	// walk the fields of the broData, making sure the broData struct has
+	// an equal number of named bro fields and bro types
+	for i := 0; i < structType.NumField(); i++ {
+		structField := structType.Field(i)
+		zeekName := structField.Tag.Get("bro")
+		zeekType := structField.Tag.Get("brotype")
+
+		//If this field is not associated with bro, skip it
+		if len(zeekName) == 0 && len(zeekType) == 0 {
+			continue
+		}
+
+		if len(zeekName) == 0 || len(zeekType) == 0 {
+			return indexMap, errors.New("incomplete bro variable")
+		}
+
+		parseTypeFields[zeekName] = parseTypeFieldInfo{
+			zeekType:             zeekType,
+			parseTypeFieldOffset: i,
+		}
+	}
+
+	for index, name := range header.Names {
+		fieldInfo, ok := parseTypeFields[name]
+		if !ok {
+			//an unmatched field which exists in the log but not the struct
+			//is not a fatal error, so we report it and move on
+			logger.WithFields(log.Fields{
+				"error":         "unmatched field in log",
+				"missing_field": name,
+			}).Info("the log contains a field with no candidate in the data structure")
+			continue
+		}
+
+		if header.Types[index] != fieldInfo.zeekType {
+			err := errors.New("type mismatch found in log")
+			logger.WithFields(log.Fields{
+				"error":         err,
+				"type in log":   header.Types[index],
```

Comment:
Minor thing but can we please remove the spaces and replace with underscores for the field names?

Reply:
Will do